core-backend/sandbox/src/env.rs

```diff
@@ -78,15 +79,21 @@ impl<E: Ext + IntoExtInfo + 'static> Environment<E> for SandboxEnvironment<E> {
         memory_pages: &BTreeMap<PageNumber, Option<Box<PageBuf>>>,
         mem_size: WasmPageNumber,
     ) -> Result<Self, BackendError<'static>> {
-        let mut later_ext = LaterExt::default();
-        later_ext.set(ext);
+        let later_ext = LaterExt::new(ext);
 
-        let mem: DefaultExecutorMemory =
-            SandboxMemory::new(mem_size.0, None).map_err(|e| BackendError {
-                reason: "Create env memory fail",
-                description: Some(format!("{:?}", e).into()),
-                gas_amount: later_ext.unset().into_gas_amount(),
-            })?;
+        let mem: DefaultExecutorMemory = match SandboxMemory::new(mem_size.0, None) {
+            Ok(mem) => mem,
+            Err(e) => {
+                return Err(BackendError {
+                    reason: "Create env memory fail",
+                    description: Some(format!("{:?}", e).into()),
+                    gas_amount: later_ext
+                        .take()
+                        .expect("method called only once with no clones around; qed")
```

Comment:
So, I forget why we have panics now all over here?

Reply:
discussed